### DISCUSSION TITLE
Unable to lazyload the plugin

### DISCUSSION THREAD

**Discussion by siduck** [mason-org/mason.nvim#122] [Q&A] [answered]
hi @williamboman I tried lazyloading the plugin at `CmdLineEnter` as I just want to use the plugin to install the  lspservers/formatter binaries but this didnt work! 

If i dont lazyload, everything works but lspconfig is very heavy on startuptime.

I tried adding `~/.local/share/nvim/mason/bin` to the path in my shell's config but that didnt work either

**Comment by williamboman** [comment, marked as answer]
Hello! Lazy-loading mason.nvim is not recommended as it is optimized to load [as little as possible](https://github.com/williamboman/mason.nvim/blob/c92cc60866930a330ddbf5e92996b46a1dab23ee/lua/mason/init.lua#L7-L16) when calling the setup function. All it does is modify your PATH + sets up the user commands. If you still insist on doing it I'm afraid you'll have to dig into the source and debug it yourself. It should work fine if you just add the bin directory to your PATH and skip setup, so not sure why it's not working for you.

**Reply by siduck** [reply]
@williamboman I know that mason.nvim is minimal at startuptime but if i dont lazyload it then i'll have to remove lazyload for lspconfig which is a heavy plugin . Also should I put this on path too?
```
~/.local/share/nvim/mason/packages
```

I tried this as of now 

```lua
vim.env.PATH = vim.env.PATH .. ":" .. vim.fn.stdpath "data" .. "/mason/bin"
```

**Comment by siduck** [comment]
@williamboman seems like its added to the path but the client doesnt attach :/ 

![image](https://user-images.githubusercontent.com/59060246/180673563-c02c0efa-0e2f-4bc0-84ee-bc1cc9d8cf74.png)

look 
 

https://user-images.githubusercontent.com/59060246/180674576-a6898ccb-936c-49cf-b31e-801cd198f680.mp4

The same thing happens when I remove the lazyloading of `mason` but my lspconfig is lazyloaded https://github.com/williamboman/mason.nvim/issues/126

nvm fixed it, i had to add `silent e %` in my lspconfig's lazyloading section.

**Reply by zhang-stephen** [reply]
I have tried lazy-loading and it works well, see [here](https://github.com/zhang-stephen/dotfiles-on-unix-like/blob/1fbbf1197fd6e68aec89e9ae8ed06fb56a8a0f15/nvim/lua/modules/lsp/init.lua#L19).

But the `setup()` seems not work... The UI didn't be changed after `setup()`...

I created a plugin, which is based on mason.nvim and nvim-lspconfig, to load language servers(DAP, formatter and linter support is in planning) from json files after mason.nvim loaded. there is nothing done when `require'mason'.setup()` except UI changing. Could you try this [nvim-lsp-loader](https://github.com/zhang-stephen/nvim-lsp-loader)?

And @williamboman, would you add this plugin to the wiki of mason.nvim?

**Reply by williamboman** [reply]
> And @williamboman, would you add this plugin to the wiki of mason.nvim?

The Wiki should be open for editing for anyone!

**Reply by siduck** [reply]
@williamboman Avoid that! some mischievous person can add misformation or unwanted content on the wiki too xD 

maybe you can try something like this https://github.com/kamiyaa/joshuto/tree/main/docs so users can PR too
